core/capabilities/targets/write_target.go

```diff
@@ -309,7 +310,44 @@ func (cap *WriteTarget) Execute(ctx context.Context, rawRequest capabilities.Cap
 	}
 
 	cap.lggr.Debugw("Transaction submitted", "request", request, "transaction", txID)
-	return capabilities.CapabilityResponse{}, nil
+
+	done := make(chan struct{})
+
+	go func() {
+		defer close(done)
+		for {
+			time.Sleep(100 * time.Millisecond)
+			status, _ := cap.cw.GetTransactionStatus(ctx, txID.String())
+
+			switch status {
+			case commontypes.Pending:
+				continue
+			case commontypes.Unconfirmed:
+				continue
+			case commontypes.Failed:
+				continue
+			case commontypes.Finalized:
+				cap.lggr.Debugw("Transaction finalized", "request", request, "transaction", txID)
+				return
+			case commontypes.Fatal:
+				cap.lggr.Debugw("Transaction fatal error", "request", request, "transaction", txID)
+				return
+			case commontypes.Unknown:
+				cap.lggr.Debugw("Transaction status unknown", "request", request, "transaction", txID)
+				return
+			}
+		}
+	}()
+
+	cap.lggr.Debugw("Waiting for transaction finalization", "request", request, "transaction", txID)
+
+	select {
+	case <-done:
+		return capabilities.CapabilityResponse{}, nil
+	case <-ctx.Done():
+		cap.lggr.Debugw("Timed out for transaction finalization", "request", request, "transaction", txID)
+		return capabilities.CapabilityResponse{}, nil
```

Comment:
Why abandon the spawned goroutine and return? If it is necessary to return without waiting for it to complete, then the receiver `WriteTarget` needs to track it somehow, so that it is not orphaned.

Reply:
I think the extra goroutine could actually be eliminated in this case. That would probably be simpler overall.
```go
dur := 500*time.Millisecond
for {
	select {
		case <-ctx.Done():
			return
		case <-time.After(dur):
			if txStatusIsTerminal(ctx) { return }
			// dur = dur*2 :shrug: 
	}
}
```

Reply:
Alternatively, pass context into the extra goroutine and honor its cancellation?